addons/website_sale/static/src/website_builder/products_item_option.js

```diff
@@ -18,20 +18,34 @@ export class ProductsItemOption extends BaseOptionComponent {
             itemSize: this.props.itemSize,
         });
 
+        this.getProductsGridTable = () => {
+            return this.env.getEditingElement().closest(".o_wsale_products_grid_table");
+        };
+
+        this.domState = useDomState(() => {
+
+            // If /shop page layout is list, do not display Size option
+            const displaySizeOption = !this.getProductsGridTable()?.classList.contains("o_wsale_products_opt_layout_list");
+
+            if(displaySizeOption && this.state.itemSize) {
+                this.addClassToTableCells(this.state.itemSize.x, this.state.itemSize.y, "selected");
+            }
+
+            return {
+                displaySizeOption,
+            }
+        });
+
         onWillStart(async () => {
             this.defaultSort = await this.props.loadInfo();
 
             // need to display "re-order" option only if shop_default_sort is 'website_sequence asc'
             this.displayReOrder = this.defaultSort[0].shop_default_sort === "website_sequence asc";
-
-            // If /shop page layout is list, do not display Size option
-            const wsale_container = this.env.getEditingElement().closest("#o_wsale_container");
-            this.displaySizeOption = !wsale_container.classList.contains("o_wsale_products_opt_layout_list");
-            this.maxWidth = parseInt(wsale_container.dataset.ppr) || 5;
+            this.maxWidth = parseInt(this.getProductsGridTable().dataset.ppr) || 5;
```

Comment:
Something is off here 🤔
The amount of columns is always 5, while it should correspond to the `ppr` value.

<img width="287" height="698" alt="image" src="https://github.com/user-attachments/assets/5f20d95d-b010-4c97-bcfe-9c71a20a96b3" />